core/server/auth/oauth.js

```diff
@@ -85,6 +124,13 @@ oauth = {
         // access token on behalf of the user who authorized the code.
         oauthServer.exchange(oauth2orize.exchange.refreshToken({userProperty: 'client'},
             exchangeRefreshToken));
+
+        /**
+         * We forward the authorization code to Ghost.org, so no need to call exchange.authorizationCode
+         * exchange.authorizationCode wraps the req and res object, which is used by passport!
+         * See exchangeAuthorizationCode!
+         */
+        oauthServer.exchange('authorization_code', exchangeAuthorizationCode);
```

Comment:
hm actually it does already validate the client
https://github.com/TryGhost/Ghost/pull/7571/files#diff-e1be96888c62e807a8c07447f0e5f164R184